SRCH-2690 remove sitelink generator code, gem, & column

MothOnMars opened the pull request:
## Summary
This PR removes everything related to the `sitelink_generator` gem (https://github.com/GSA/sitelink_generator), which is no longer needed in production. 
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review:

#### Functionality Checks

- [x] Code is functional. - smoke-tested locally

- [x] Automated checks pass, if applicable. If Code Climate checks do not pass, explain reason for failures:
Both failures are false positives:

- `Possible DoS Vulnerability in Active Record PostgreSQL adapter` does not apply to us, as we do not use Postgres.
- `Possible Open Redirect in Host Authorization Middleware ` is actually fixed in 6.0.3.5 (https://groups.google.com/g/rubyonrails-security/c/zN_3qA26l6E), but Github continues to flag it erroneously.

- [x] If your changes will be tested manually, you have run `bundle update` and committed your changes to Gemfile.lock.
 
- [x] You have merged the latest changes from the target branch (usually `master` or `main`) into your branch.
 
- [x] If your target branch is NOT `master` or `main`, specify the reason:
 
- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release #.#.#** matching the release number
 
- [x] You have squashed your commits into a single commit (exception: your PR includes commits with formatting-only changes, such as required by Rubocop)

- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket
 
#### Process Checks

- [x] You have specified an "Assignee", and if necessary, additional reviewers

MothOnMars commented:
O hai - turning this back into a draft while I make CodeClimate happy...

lorawoodford commented on db/seeds/search_module.rb:
```diff
@@ -18,7 +18,6 @@
 SearchModule.create(tag: 'OSPEL', display_name: 'Spelling Suggestions (Search.gov Images)')
 SearchModule.create(tag: 'AWEB', display_name: 'Web Results Only (Azure)')
 SearchModule.create(tag: 'FRDOC', display_name: 'Federal Register Documents')
-SearchModule.create(tag: 'DECOR', display_name: 'Value-added Links for SEC Filings')
```
Just a comment that this should bump https://cm-jira.usa.gov/browse/SRCH-2695 up in the queue. :-)

lorawoodford commented:
Two questions, both likely more process/expectation questions, more so than blockers to this.  What's here LGTM.

On process:  I know the codeclimate issues are false positives.  Who in this process - developer or reviewer - clicks "Approve"?

On expectations: Is there any expectation to clean up after ourselves in the db to ensure the contents of the `search_modules` db table matches https://search.gov/admin-center/analytics/module-codes.html?  (Seems like no -- they're not a match now?). Essentially, do we need to bother ourselves with rolling this back: [db/migrate/20150111220313_add_sitelink_search_module.rb](https://github.com/GSA/search-gov/blob/master/db/migrate/20150111220313_add_sitelink_search_module.rb)?

Thanks for humoring the questions!

MothOnMars commented:
Good q's: 

> Two questions, both likely more process/expectation questions, more so than blockers to this. What's here LGTM.
> 
> On process: I know the codeclimate issues are false positives. Who in this process - developer or reviewer - clicks "Approve"?

We've been somewhat inconsistent about this in the past, but I think it's appropriate for the reviewer to hit "Approve". I've added a line about this in our Confluence doc: https://cm.usa.gov/confluence/display/SRCH/How+to+Write+and+Work+on+Stories#HowtoWriteandWorkonStories-State:CodeReview

> 
> On expectations: Is there any expectation to clean up after ourselves in the db to ensure the contents of the `search_modules` db table matches https://search.gov/admin-center/analytics/module-codes.html? (Seems like no -- they're not a match now?). Essentially, do we need to bother ourselves with rolling this back: [db/migrate/20150111220313_add_sitelink_search_module.rb](https://github.com/GSA/search-gov/blob/master/db/migrate/20150111220313_add_sitelink_search_module.rb)?
> 
> Thanks for humoring the questions!

A super admin can take care of that cleanup at some point (https://search.usa.gov/admin/search_modules), but for now we can leave it. The `DECOR` module will still appear in site analytics for the next 18 months(+), so it can serve as reference. I've also added a note to https://cm-jira.usa.gov/browse/SRCH-2695 that we should flag a number of modules listed in https://search.gov/admin-center/analytics/module-codes.html as deprecated.

lorawoodford reviewed (approved):
LGTM
